src/components/Api.js

```diff
@@ -0,0 +1,25 @@
+export default class Api {
+  constructor({ url, method, headers, body }) {
+    this._url = url;
+    this._method = method || "GET";
+    this._headers = headers;
+    this._body = body;
+  }
+
+  handleFetch() {
+    return fetch(this._url, {
+      method: this._method,
+      headers: this._headers,
+      body: this._body,
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.